syntax = "proto3";
package sqlpkgs;




import "model/database_delivery/sql_package_version.proto";

/*
UpdateSQLPackage请求
*/
message UpdateSQLPackageRequest {
message UpdateSqlpkg {
    /*
    名称
    */
    string name = 1 ;
    /*
    描述
    */
    string memo = 2 ;
    /*
    程序包Id
    */
    string appPackageId = 3 ;
}
        
    
    /*
    SQL包ID
    */
    string pkgId = 1 ;
    /*
    更新的SQL包body
    */
    UpdateSqlpkg updateSqlpkg = 2 ;
}

/*
UpdateSQLPackage返回
*/
message UpdateSQLPackageResponse {
    /*
    SQL包下的版本列表
    */
    repeated database_delivery.SQLPackageVersion versionList = 1 ;
    /*
    SQL包ID
    */
    string id = 2 ;
    /*
    名称
    */
    string name = 3 ;
    /*
    SQL包的数据库类型
    */
    string dbType = 4 ;
    /*
    描述
    */
    string memo = 5 ;
    /*
    创建者
    */
    string creator = 6 ;
    /*
    ctime
    */
    int64 ctime = 7 ;
    /*
    mtime
    */
    int64 mtime = 8 ;
    /*
    程序包Id
    */
    string appPackageId = 9 ;
    /*
    文件仓库包ID
    */
    string repoPackageId = 10 ;
}

/*
UpdateSQLPackageApi返回
*/
message UpdateSQLPackageResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    sqlpkgs.UpdateSQLPackageResponse data = 4 ;
}

